pkgs/by-name/ko/koboldcpp/package.nix

```diff
@@ -106,14 +119,31 @@ effectiveStdenv.mkDerivation (finalAttrs: {
   env.NIX_LDFLAGS = lib.concatStringsSep " " (finalAttrs.darwinLdFlags ++ finalAttrs.metalLdFlags);
 
   env.NIX_CFLAGS_COMPILE =
-    lib.optionalString (march != "") "-march=${march}" + lib.optionalString (mtune != "") "-mtune=${mtune}";
+    let
+      inherit (stdenv.hostPlatform)
+        isx86_64
+        isAarch64
+        isLinux
+        isDarwin
+        ;
+    in
+    lib.concatStringsSep " " [
+      (lib.optionalString isx86_64 "-march=x86-64")
+      (lib.optionalString (isAarch64 && isLinux) "-march=armv8.7-a")
+      (lib.optionalString (mtune != "") "-mtune=${mtune}")
```

Comment:
This is set to `""` by default, so it's not actually used in the package set. If the user wishes to reconfigure this, they can use `.override = { stdenv = myCustomStdenv; }` or `.overrideAttrs (oldAttrs: { NIX_CFLAGS_COMPILE = ... ; })`